python/lsst/analysis/tools/tasks/injectedObjectAnalysis.py

```diff
@@ -0,0 +1,64 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("InjectedObjectAnalysisConfig", "InjectedObjectAnalysisTask")
+
+from lsst.pipe.base import connectionTypes as ct
+from lsst.skymap import BaseSkyMap
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+
+class InjectedObjectAnalysisConnections(
+    AnalysisBaseConnections,
+    dimensions=("skymap", "tract"),
+    defaultTemplates={
+        "outputName": "matched_injected_deepCoadd_catalog_tract_injected_objectTable_tract",
```

Comment:
Does injected need to be in this name twice? I am not sure where it is actually named and guess that it is not on this ticket.

Reply:
I believe this comes from [DM-41210](https://rubinobs.atlassian.net/browse/DM-41210). The format for the connection name is "matched_{catalog1}_{catalog2}". Here both catalogs being matched happen to have "injected_" in the name.

Reply:
That's the default connection name format for matched tables. Attempting to shorten the name would just make it harder to figure out which tables were matched.